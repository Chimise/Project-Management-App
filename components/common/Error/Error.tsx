import React from 'react';
import cn from 'classnames';

interface ErrorProps {
    message?: string;
    onRetry: () => void;
    className?: string
}

const Error = ({message, onRetry, className}: ErrorProps) => {
    return <div className={cn('w-full flex flex-col justify-center items-center space-y-4', className)}>
        <p className='text-sm text-gray-700 text-light'>{message ? message : 'An error occured, please try again'}</p>
        <button onClick={onRetry} className='py-2 px-3 rounded-md bg-black/70 focus:ouline-none hover:bg-black'>Retry</button>
    </div>
}

export default Error;