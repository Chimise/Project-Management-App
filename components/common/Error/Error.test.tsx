import "@testing-library/jest-dom";
import {
  render,
  user,
} from "../../../test/test-utils";
import router, {
  mockUseRouter,
} from "../../../test/router";
import Error from './Error';

beforeEach(() => {
  jest.useFakeTimers();
  jest.clearAllMocks();
});

afterEach(() => {
  jest.useRealTimers();
});

mockUseRouter();

test('The error messages is displayed correctly', () => {
    const error = 'An error occured'
    const handler = jest.fn();
    const {getByText} = render(<Error onRetry={handler} message={error} /> );
    expect(getByText(error)).toBeInTheDocument();
})

test('Check that a default error message is shown', () => {
    const handler = jest.fn();
    const {getByText} = render(<Error onRetry={handler} />);
    expect(getByText(/error/i)).toBeInTheDocument();
})

test('The onretry handler is called when the retry button is clicked', async () => {
    const handler = jest.fn();
    const {getByRole} = render(<Error onRetry={handler} />);
    await user.click(getByRole('button', {name: 'Retry'}));
    expect(handler).toHaveBeenCalled();
})

test('It renders unchanged', () => {
    const handler = jest.fn();
    const {container} = render(<Error onRetry={handler} />);
    expect(container.firstChild).toMatchSnapshot();
})