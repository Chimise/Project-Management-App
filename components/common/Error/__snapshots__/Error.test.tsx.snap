// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It renders unchanged 1`] = `
<div
  class="w-full flex flex-col justify-center items-center space-y-4"
>
  <p
    class="text-sm text-gray-700 text-light"
  >
    An error occured, please try again
  </p>
  <button
    class="py-2 px-3 rounded-md bg-black/70 focus:ouline-none hover:bg-black"
  >
    Retry
  </button>
</div>
`;
